src/v/archival/probe.cc

```diff
@@ -67,6 +74,40 @@ ntp_level_probe::ntp_level_probe(
       });
 }
 
+void ntp_level_probe::setup_public_metrics(const model::ntp& ntp) {
+    namespace sm = ss::metrics;
+
+    auto ns_label = ssx::metrics::make_namespaced_label("namespace");
+    auto topic_label = ssx::metrics::make_namespaced_label("topic");
+    auto partition_label = ssx::metrics::make_namespaced_label("partition");
+    const std::vector<sm::label_instance> labels = {
+      ns_label(ntp.ns()),
+      topic_label(ntp.tp.topic()),
+      partition_label(ntp.tp.partition()),
+    };
+
+    auto aggregate_labels = std::vector<sm::label>{
```

Comment:
We might want to revisit this at some point to maintain some shard-wide counters, rather than doing per-ntp counters that will get summed up at scrape time.